File: Dec2020/9Dec/InvertBinaryTree.java
/*
Problem Name: 226. Invert Binary Tree
Problem Link: https://leetcode.com/problems/invert-binary-tree/
*/
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        return invert(root);
    }
    TreeNode invert(TreeNode root){
        if(root == null)
            return null;
        TreeNode r = invert(root.right), l = invert(root.left);
        root.left = r;
        root.right = l;
        return root;
    }
}

File: Dec2020/9Dec/InvertBinaryTree.cpp
/*
Problem name: 226. Invert Binary Tree
Problem link: https://leetcode.com/problems/invert-binary-tree/
*/

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if(!root)
            return root;
        
        TreeNode* temp1;
        TreeNode* temp2;
        
        temp1=invertTree(root->left);
        temp2=invertTree(root->right);
        
        root->right=temp1;
        root->left=temp2;
        
        return root;
    }
};
